### PR TITLE
feat(trace-view): Add substring search to trace view

### PR DESCRIPTION
**Pull request by Zylphrex**
This change implements the ability to search the trace by transaction name
through a substring condition.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/performance/traceDetails/content.tsx [hunk 1]
```diff
@@ -55,6 +70,53 @@ class TraceDetailsContent extends React.Component<Props> {
...
+        this.setState({
+          filteredTransactionIds: undefined,
+        });
+      }
+      return;
+    }
+
+    // TODO(tonyx): this should perform a fuzzy search
+    const matched = filterTrace(trace, ({transaction}) =>
+      transaction.includes(searchQuery)
```
Comment (wmak) [inline src/sentry/static/sentry/app/views/performance/traceDetails/content.tsx:92]: nit: I know we're switching to fuzzy, but could this be case insensitive?

Comment (wmak) [reply src/sentry/static/sentry/app/views/performance/traceDetails/content.tsx:92]: nit2: (maybe not a nit though, i just know we're adding fuzzy search so iffy on whether we need to do in this PR) But we need to search the transaction.op too to match how span's do this

Comment (Zylphrex) [reply src/sentry/static/sentry/app/views/performance/traceDetails/content.tsx:92]: I can update this to do a case insensitive search on the transaction op and name until fuzzy search is implemented.

Comment (Zylphrex) [reply src/sentry/static/sentry/app/views/performance/traceDetails/content.tsx]: Updated in 43641fa

**Review by wmak — APPROVED** [review]

**Pull request merged**